apps/verticalhome/js/app.js

```diff
@@ -157,14 +204,43 @@
       this.renderGrid();
     },
 
+
+    /**
+     * We need to do some refreshing thing after l10n is ready.
+     *
+     * @memberof LockScreen
+     * @this {LockScreen}
+     */
+    l10nInit: function ls_l10nInit() {
+      this.l10nready = true;
+      // The default one is 12 hour.
+      this.timeFormat = window.navigator.mozHour12 ?
+        navigator.mozL10n.get('shortTimeFormat12') :
+        navigator.mozL10n.get('shortTimeFormat24');
+      this.refreshClock(new Date());
+
+    },
+
+
+    refreshClock: function ls_refreshClock(now) {
+      var f = new navigator.mozL10n.DateTimeFormat();
+      var _ = navigator.mozL10n.get;
+
+      var timeFormat = this.timeFormat.replace('%p', '<span>%p</span>');
+      var dateFormat = _('longDateFormat');
+      this.clockTime.innerHTML = f.localeFormat(now, timeFormat);
```

Comment:
writing to innerHTML will cause trouble on a security audit. It is okay to set innerHTML to the empty string to delete elements, but setting this to a string, especially something that is not a string literal is considered a risk of bugs. You'll have to change this before we can land it.